hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCallHandler.java

```diff
@@ -89,23 +82,24 @@ public void preHandle(@NonNull final PreHandleContext context) {
 
     @Override
     public void pureChecks(@NonNull TransactionBody txn) throws PreCheckException {
-        final var op = txn.contractCallOrThrow();
-        mustExist(op.contractID(), INVALID_CONTRACT_ID);
+        try {
+            final var op = txn.contractCallOrThrow();
+            mustExist(op.contractID(), INVALID_CONTRACT_ID);
 
-        final var intrinsicGas = gasCalculator.transactionIntrinsicGasCost(
-                Bytes.wrap(op.functionParameters().toByteArray()), false);
-        validateTruePreCheck(op.gas() >= intrinsicGas, INSUFFICIENT_GAS);
+            final var intrinsicGas = gasCalculator.transactionIntrinsicGasCost(
+                    Bytes.wrap(op.functionParameters().toByteArray()), false);
+            validateTruePreCheck(op.gas() >= intrinsicGas, INSUFFICIENT_GAS);
+        } catch (@NonNull final Exception e) {
+            bumpExceptionMetrics(CONTRACT_CALL, e);
+            throw e;
+        }
     }
 
-    @NonNull
     @Override
-    public Fees calculateFees(@NonNull final FeeContext feeContext) {
-        requireNonNull(feeContext);
-        final var op = feeContext.body();
-        return feeContext
-                .feeCalculatorFactory()
-                .feeCalculator(SubType.DEFAULT)
-                .legacyCalculate(sigValueObj ->
-                        usageEstimator.getContractCallTxFeeMatrices(CommonPbjConverters.fromPbj(op), sigValueObj));
+    protected /*abstract*/ @NonNull FeeData getFeeMatrices(
```

Comment:
`/*abstract*/`?

Reply:
It can't actually be `abstract` because then all inheritors would have to implement it, which isn't the case already (`EthereumTransaction` doesn't need it.). But I kind of want to make it clear it is the subclass's responsibility to implement it if needed (if the subclass doesn't override `calculateFees`).  So there's isn't really a way to mark it as "kind of `abstract`". ...